SOURCES/etc/xapi.d/plugins/lvm.py

```diff
@@ -0,0 +1,235 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import errno
+import json
+import sys
+
+import XenAPIPlugin
+
+from xcpngutils import configure_logging, run_command, error_wrapped, strtobool
+
+@error_wrapped
+def list_physical_volumes(session, args):
+    try:
+        command = ['pvs', '--noheadings', '--units', 'B', '--nosuffix']
+        result = run_command(command)
+
+        # Format:
+        # PV device name
+        # VG name
+        # Format
+        # Attributes
+        # Physical size
+        # Physical free size
+
+        pvs_result = {}
+        for index, line in enumerate(result['stdout'].splitlines()):
+            fields = line.split()
+            if len(fields) != 6:
+                raise Exception('Unexpected pv result at index {}.'.format(index))
```

Comment:
Wouldn't including the full line in the exception message make it easier to support/debug?